packages/vite/src/node/preview.ts

```diff
@@ -48,12 +48,28 @@ export async function preview(
 
   const options = config.server
   const hostname = resolveHostname(serverOptions.host ?? options.host)
-  const port = serverOptions.port ?? 5000
+  let port = serverOptions.port ?? 5000
   const protocol = options.https ? 'https' : 'http'
   const logger = config.logger
   const base = config.base
 
+  const onError = (e: Error & { code?: string }) => {
+    if (e.code === 'EADDRINUSE') {
+      logger.info(`Port ${port} is in use, trying another one...`)
+      httpServer.listen(++port, hostname.host)
+    } else {
+      httpServer.removeListener('error', onError)
+      throw e
+    }
+  }
+
+  if (serverOptions.port === undefined) {
```

Comment:
**suggestion:** We could just double-equals check for null here, so just to be sure that we also just check for `null` or `undefined`
```suggestion
  if (serverOptions.port == null) {
```

Reply:
`!serverOptions.port`, is that okay?
```suggestion
  if (!serverOptions.port) {
```

Reply:
I think yes, cause a port of `0` is very unlikely 😉